### PR TITLE
CAPZ: Release v28.0.0.

### PR DESCRIPTION
**Pull request by Gacko**
### Checklist

- [x] Roadmap issue created
- [x] Release uses latest stable Flatcar
- [x] Release uses latest Kubernetes patch version

### Triggering E2E tests

To trigger the E2E test for each new Release added in this PR, add a comment with the following:

`/run releases-test-suites PREVIOUS_RELEASE=27.0.0`

If you want to trigger conformance tests, you can do so by adding a comment similar to the following:

`/run conformance-tests PROVIDER=azure RELEASE_VERSION=v28.0.0`

For more details see the [README.md](/README.md#running-tests-against-prs).

### CODE REVIEW

**Comment by Gacko** [comment]
/run conformance-tests PROVIDER=azure RELEASE_VERSION=v28.0.0

**Comment by Gacko** [comment]
I see this output in the Release Tests Suites:

```
Basic upgrade test upgrade successfully finishes control plane nodes rolling update if it is needed
/app/internal/upgrade/upgrade.go:176
  [SKIPPED] in [It] - /app/internal/upgrade/upgrade.go:192 @ 08/12/24 23:03:02.284
S [SKIPPED] [0.055 seconds]
Basic upgrade test upgrade [It] successfully finishes control plane nodes rolling update if it is needed
/app/internal/upgrade/upgrade.go:176

  [SKIPPED] Control plane resource generation did not change, skipping rolling update test
  In [It] at: /app/internal/upgrade/upgrade.go:192 @ 08/12/24 23:03:02.284
```

I assume this means the tests are not waiting for all nodes to be replaced because it thinks there has not been any change that could cause a node roll, right? Well, actually there is a good reason: This is an upgrade von Kubernetes v1.27 to v1.28.

Therefore the nodes roll while tests just go on and finally fail.

@giantswarm/team-tinkerers WDYT?

**Comment by AverageMarcus** [comment]
> I see this output in the Release Tests Suites:
> 
> ```
> Basic upgrade test upgrade successfully finishes control plane nodes rolling update if it is needed
> /app/internal/upgrade/upgrade.go:176
>   [SKIPPED] in [It] - /app/internal/upgrade/upgrade.go:192 @ 08/12/24 23:03:02.284
> S [SKIPPED] [0.055 seconds]
> Basic upgrade test upgrade [It] successfully finishes control plane nodes rolling update if it is needed
> /app/internal/upgrade/upgrade.go:176
> 
>   [SKIPPED] Control plane resource generation did not change, skipping rolling update test
>   In [It] at: /app/internal/upgrade/upgrade.go:192 @ 08/12/24 23:03:02.284
> ```
> 
> I assume this means the tests are not waiting for all nodes to be replaced because it thinks there has not been any change that could cause a node roll, right? Well, actually there is a good reason: This is an upgrade of Kubernetes v1.27 to v1.28.
> 
> Therefore the nodes roll while tests just go on and finally fail.
> 
> @giantswarm/team-tinkerers WDYT?

We didn't write this test. Could you please take a look at the git blame and work with whoever implemented it to figure out what's going wrong?

**Comment by Gacko** [comment]
Oh, true. Sorry to bother you then and thanks for pointing me in the right direction!

Ciao, @weseven! 🙂

https://github.com/giantswarm/cluster-test-suites/pull/422

**Comment by weseven** [comment]
> Oh, true. Sorry to bother you then and thanks for pointing me in the right direction!
> 
> Ciao, @weseven! 🙂
> 
> [giantswarm/cluster-test-suites#422](https://github.com/giantswarm/cluster-test-suites/pull/422)

How is it possible that the control plane resource does not change if we're changing kubernetes version? :thinking: 
I've tested this in CAPA and the resource generation changed as expected.
Do you have a link to the failed pipelinerun?

**Comment by Gacko** [comment]
The pipeline does not need to fail for this. If you're lucky, all the tests still pass while the cluster is rolling. I have no idea, but maybe the Kubernetes version (only change here) is not part of the resource we are checking in CAPZ?

**Comment by Gacko** [comment]
Could you please review this PR anyway? We can discuss the test issue somewhere else. I mean, the tests passed, even though they ran while the cluster was rolling.

**Comment by Gacko** [comment]
/run conformance-tests PROVIDER=azure RELEASE_VERSION=v28.0.0

**Review by weseven — APPROVED** [review]

**Comment by weseven** [comment]
> The pipeline does not need to fail for this. If you're lucky, all the tests still pass while the cluster is rolling. I have no idea, but maybe the Kubernetes version (only change here) is not part of the resource we are checking in CAPZ? Or we are checking it too fast and the cluster chart has not updated the resource yet at the moment of checking?
> 
> Could you please review this PR anyway? We can discuss the test issue somewhere else. I mean, the tests passed, even though they ran while the cluster was rolling.

We are checking after the whole new app version has been successfully deployed, and the new app updates the kubeadmcontrolplane resource (no capi controllers involved), so it's pretty weird. 
Maybe it could be some caching on the k8sclient used to retrieve the resource, I will need to replicate this.

Anyway, approved the PR, lgtm!

File: azure/v28.0.0/release.diff [hunk 1]
```diff
@@ -0,0 +1,118 @@
+apiVersion: release.giantswarm.io/v1alpha1				apiVersion: release.giantswarm.io/v1alpha1
```
Comment (AndiDog) [inline azure/v28.0.0/release.diff:1]: Diff looks strange

**Review by AndiDog — APPROVED** [review]
~Apart from the strange diff file,~ it looks okay

File: azure/v28.0.0/release.diff (hunk 1 above)
Comment (AndiDog) [reply azure/v28.0.0/release.diff:1]: Scratch that – it's a side-by-side which looks really nasty on a small screen with the GitHub PR open

**Review by AndiDog — APPROVED** [review]

File: azure/v28.0.0/release.diff (hunk 1 above)
Comment (Gacko) [reply azure/v28.0.0/release.diff:1]: How exactly does it look strange for you? Actually I only did `diff <old> <new>` and I agree, depending on how you're reading it, tabs get display correctly or not. It sucks... but still better than nothing.

**Pull request merged**